add task solution

Anton777148 opened the pull request:
[DEMO LINK](https://Anton777148.github.io/react_todo-app-loading-todos/)

Anton777148 committed:
add task solution

volodymyr-soltys97 commented on src/components/TodoList.tsx:
```diff
@@ -0,0 +1,330 @@
+      setFilterBy(filter);
+    };
+
+  const filters = [
+    { type: FilterType.All, label: 'All', cy: 'FilterLinkAll' },
+    { type: FilterType.Active, label: 'Active', cy: 'FilterLinkActive' },
+    {
+      type: FilterType.Completed,
+      label: 'Completed',
+      cy: 'FilterLinkCompleted',
+    },
+  ];
```
Don't need to create additional array, use `Object.values(FilterType)`

volodymyr-soltys97 reviewed (changes requested):
Good job 👍 
Let's improve your code
You need to split the logic into more components, for example `Header, Footer`

Anton777148 committed:
add task solution

etojeDenys commented on src/App.tsx:
```diff
@@ -1,10 +1,59 @@
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
```
this state looks redundant

etojeDenys commented on src/App.tsx:
```diff
@@ -1,10 +1,59 @@
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [selectedLink, setSelectedLink] = useState(FilterType.All);
+  const [errorButton, setErrorButton] = useState(false);
```
this too

etojeDenys commented on src/App.tsx:
```diff
@@ -1,10 +1,59 @@
+  const [selectedLink, setSelectedLink] = useState(FilterType.All);
+  const [errorButton, setErrorButton] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todosCounter, setTodosCounter] = useState(0);
```
same here

etojeDenys commented on src/App.tsx:
```diff
@@ -1,10 +1,59 @@
+      });
+  }, []);
+
+  useEffect(() => {
+    switch (selectedLink) {
+      case FilterType.active:
+        setTodos(allTodos.filter(todo => !todo.completed));
+        break;
+      case FilterType.completed:
+        setTodos(allTodos.filter(todo => todo.completed));
+        break;
+      default:
+        setTodos(allTodos);
+        break;
+    }
+  }, [selectedLink, allTodos]);
```
it's just can be the var
```
const filteredTodos = filterTodos(selectedLink)
```

Anton777148 committed:
add task solution

anastasiiavorobiova commented on src/App.tsx:
```diff
@@ -1,10 +1,55 @@
 export const App: React.FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [selectedLink, setSelectedLink] = useState(FilterType.All);
+  const [errorButton, setErrorButton] = useState(false);
```
```suggestion

```

anastasiiavorobiova commented on src/App.tsx:
```diff
@@ -1,10 +1,55 @@
+  const [selectedLink, setSelectedLink] = useState(FilterType.All);
+  const [errorButton, setErrorButton] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [todosCounter, setTodosCounter] = useState(0);
```
```suggestion

```

anastasiiavorobiova commented on src/App.tsx:
```diff
@@ -14,190 +59,54 @@ export const App: React.FC = () => {
+        className={classNames(
+          'notification is-danger is-light has-text-weight-normal',
+          {
+            hidden: errorButton || errorMessage.length === 0,
```
```suggestion
            hidden: errorMessage.length === 0,
```

anastasiiavorobiova commented on src/App.tsx:
```diff
@@ -14,190 +59,54 @@ export const App: React.FC = () => {
-      {/* DON'T use conditional rendering to hide the notification */}
-      {/* Add the 'hidden' class to hide the message smoothly */}
       <div
         data-cy="ErrorNotification"
```
Consider creating a separate component for the error

anastasiiavorobiova commented on src/components/TodoItem.tsx:
```diff
@@ -0,0 +1,159 @@
+      .then(() => {
+        const filtered = allTodos.filter(todoItem => todoItem.id !== todoId);
+
+        // setTodos([...filtered]);
```
```suggestion

```

anastasiiavorobiova reviewed (changes requested):
Great work! Almost done

Anton777148 committed:
Update src/App.tsx

in the tests: should have todosCounter with a number of not completed todos
4.4s

AssertionError: Timed out retrying after 4000ms: expected '<span.todo-count>' to have text '2 items left', but the text was ''

Co-authored-by: Anastasiia Vorobiova <135232001+anastasiiavorobiova@users.noreply.github.com>

Anton777148 committed:
add task solution

anastasiiavorobiova reviewed (changes requested):
Consider fixing errors here
If you have deleted a variable you can't reference it
Feel free to ask for help in the chat

Anton777148 committed:
add task solution

Anton777148 committed:
add task solution--no-verify

Anton777148 committed:
add task solution

volodymyr-soltys97 reviewed (changes requested):
Hi, this element looks redundant, need to remove it and make a deploy again
<img width="312" alt="image" src="https://github.com/user-attachments/assets/fc2ab20d-4367-4a46-9b1a-f684134be85e" />

Anton777148 committed:
add task solution

volodymyr-soltys97 reviewed (approved):
Great! 🔥
